aptos-move/dynamic-transaction-composer/src/builder.rs

```diff
@@ -0,0 +1,498 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ArgumentOperation, BatchArgument, BatchedFunctionCall, PreviousResult};
+use anyhow::{anyhow, bail, Result};
+use move_binary_format::{
+    access::ModuleAccess,
+    binary_views::BinaryIndexedView,
+    file_format::{Ability, AbilitySet, FunctionDefinition, FunctionHandle, SignatureToken},
+    CompiledModule,
+};
+use move_core_types::{
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, StructTag, TypeTag},
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{collections::BTreeMap, str::FromStr};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+/// Arguments for each function.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum BatchArgumentType {
+    Raw,
+    Signer,
+    PreviousResult,
+}
+
+#[wasm_bindgen(js_name = BatchArgument)]
+/// Arguments for each function. Wasm bindgen only support C-style enum so use option to work around.
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct BatchArgumentWASM {
+    ty: BatchArgumentType,
+    raw: Option<Vec<u8>>,
+    signer: Option<u16>,
+    previous_result: Option<PreviousResult>,
+}
+
+#[wasm_bindgen]
+pub struct BatchedFunctionCallBuilder {
```

Comment:
As we call the thing now `transaction composer`, the type should be `struct TransactionComposer`.

Similarly, we should make all the naming here more consistent and concise. There was a discussion that 'batch' is not the intended name for this, so we should remove it. Some possible naming:

```
struct ArgumentType;
struct Argument;
struct TransactionComposer;
fn single_signer;
fn multi_signer;
fn add_call;
fn compile;
```

Reply:
Please address